relay-server/src/actors/upstream.rs

```diff
@@ -223,14 +245,27 @@ struct UpstreamRequest {
 }
 
 pub struct UpstreamRelay {
+    /// backoff policy for the registration messages
     backoff: RetryBackoff,
+    /// backoff policy for the network outage message
+    network_status_backoff: RetryBackoff,
+    /// from this instant forward we only got network errors on all our http requests
+    /// (any request that is sent without causing a network error resets this back to None)
     first_error: Option<Instant>,
     config: Arc<Config>,
     auth_state: AuthState,
     max_inflight_requests: usize,
     num_inflight_requests: usize,
+    /// queue holding the high priority messages to be sent
```

Comment:
_nit_: Are these comments really necessary? In my eyes, the signature of these fields is already pretty self-explanatory.

Reply:
ok

Reply:
**Comment to previous change request (line 86...89)** 

The test you are talking about is under the is_client_error(), it is marked as redundant because of checking is_client_error().

The `is_client_error()` is there since the original implementation (before retries) and I thought it would make sense to keep it since there is little sense in ever trying an authentication after you get a 4xx result (regardless of the body).

Let's double check that this is indeed what we want, the fix is simple enough ( just remove the is_clent_error() test and leave only the next test).